src/contracts/atlas/Escrow.sol

```diff
@@ -313,6 +294,37 @@ abstract contract Escrow is AtlETH {
         return (result, gasLimit);
     }
 
+    /// @notice Validates a SolverOperation's deadline against the current block.
+    /// @param solverOp The SolverOperation being validated.
+    /// @param dConfig DApp configuration data, including solver gas limits and operation parameters.
+    /// @return result Updated result flags after performing the validation checks, including any new errors
+    function _validateSolverOpDeadline(
+        SolverOperation calldata solverOp,
+        DAppConfig memory dConfig
+    )
+        internal
+        view
+        returns (uint256 result)
+    {
+        if (solverOp.deadline != 0 && block.number > solverOp.deadline) {
+            result |= (
+                1
+                    << uint256(
+                        dConfig.callConfig.allowsTrustedOpHash()
+                            ? uint256(SolverOutcome.DeadlinePassedAlt)
+                            : uint256(SolverOutcome.DeadlinePassed)
+                    )
+            );
+            return result;
+        }
+
+        uint256 lastAccessedBlock = accessData[solverOp.from].lastAccessedBlock;
+
+        if (lastAccessedBlock == block.number) {
```

Comment:
```suggestion
        if (lastAccessedBlock >= block.number) {
```
This is how its checked in `Sorter._verifySolverEligibility()`. Maybe safest to use here as well, or change Sorter to `==`

Reply:
Cool, that works